src/components/step-flow/step-flow.pw.tsx

```diff
@@ -0,0 +1,275 @@
+    }) => {
+      await mount(
+        <StepFlowComponent totalSteps={totalSteps} currentStep={currentStep} />
+      );
+
+      if (totalSteps >= currentStep) {
+        await expect(stepFlowLabel(page)).toHaveText(
+          `Step ${currentStep} of ${totalSteps}`
+        );
+      } else {
+        await expect(stepFlowLabel(page)).toHaveText(
+          `Step ${totalSteps} of ${totalSteps}`
+        );
+      }
+    });
+  });
+
+  ([
+    [1, 1],
+    [2, 1],
+    [2, 3],
+  ] as [Steps, Steps][]).forEach(([totalSteps, currentStep]) => {
+    test(`should render the correct visually hidden title when the 'totalSteps' prop is passed as ${totalSteps} and 'currentStep' is ${currentStep}, and the category and title props are also passed`, async ({
+      mount,
+      page,
+    }) => {
+      const category = "foo";
+      const title = "bar";
+
+      await mount(
+        <StepFlowComponent
+          totalSteps={totalSteps}
+          currentStep={currentStep}
+          category={category}
+          title={title}
+        />
+      );
+
+      if (totalSteps >= currentStep) {
+        await expect(stepFlowVisuallyHiddenTitleText(page)).toHaveText(
+          `${category}. ${title}. Step ${currentStep} of ${totalSteps}.`
+        );
+      } else {
+        await expect(stepFlowVisuallyHiddenTitleText(page)).toHaveText(
+          `${category}. ${title}. Step ${totalSteps} of ${totalSteps}.`
+        );
+      }
+    });
+  });
+
+  ([1, 2, 3, 4, 5, 6, 7, 8] as Steps[]).forEach((totalSteps) => {
+    test(`should render the correct amount of progress indicators when the 'totalSteps' prop is passed as ${totalSteps}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <StepFlowComponent totalSteps={totalSteps} showProgressIndicator />
+      );
+
+      expect(await stepFlowProgressIndicator(page).count()).toEqual(totalSteps);
+    });
+  });
+
+  test("should render the correct element when 'showCloseIcon' is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<StepFlowComponent showCloseIcon />);
+
+    await expect(stepFlowDismissIcon(page)).toBeVisible();
+  });
+
+  test("background colour token should be correct when 'showCloseIcon' is true and the indicator is in progress", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <StepFlowComponent currentStep={1} totalSteps={1} showProgressIndicator />
+    );
+
+    const backgroundColorToken = await getDesignTokensByCssProperty(
+      page,
+      stepFlowProgressIndicator(page),
+      "background-color"
+    );
+
+    expect(backgroundColorToken[0]).toBe("--colorsUtilityYin090");
+  });
+
+  test("background colour token should be correct when 'showCloseIcon' is true and the indicator is completed", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <StepFlowComponent currentStep={2} totalSteps={3} showProgressIndicator />
+    );
+
+    const backgroundColorToken = await getDesignTokensByCssProperty(
```

Comment:
suggestion(non-blocking): Same again. You may not need this await here.

```suggestion
    const backgroundColorToken = getDesignTokensByCssProperty(
```

Reply:
I think we do need an await here as its a promise, without it `backgroundColorToken` returns `undefined`